app/src/main/java/space/taran/arkmemo/data/repositories/TextNotesRepository.kt

```diff
@@ -17,69 +20,156 @@ import kotlin.io.path.getLastModifiedTime
 
 class TextNotesRepository @Inject constructor() {
 
-    @Inject @ApplicationContext lateinit var context: Context
+    @Inject
+    @ApplicationContext
+    lateinit var context: Context
 
-    fun saveNote(note: TextNote?) {
+    fun saveNote(note: TextNote?, rootResourceId: ResourceId? = null): Long {
+        val versionStorage = VersionStorage(getPath()!!)
         if (note != null) {
             val path = getPath()
             if (path != null) {
                 Files.list(path)
-                createTextNoteFile(
-                    path,
-                    JsonParser.parseNoteToJson(note.content),
-                )
+                val newResourceId =
+                    createTextNoteFile(
+                        path,
+                        JsonParser.parseNoteContentToJson(note.content),
+                    )!!
+                if (newResourceId < 0) {//Error creating TextNote file:
```

Comment:
_(Future work)_

Good point, but ideally id must always come from the lib and this way we could not worry about wrong values.

In Rust lib it is always unsigned (`u64`): https://github.com/ARK-Builders/arklib/blob/main/src/id.rs#L51

In Android bindings it is plain Long, but we can rely on it to be unsigned: https://github.com/ARK-Builders/arklib-android/blob/3334850384f84fe8d5f0545a221cc1fbc14e722e/lib/src/main/java/space/taran/arklib/Lib.kt#L50